### PR TITLE
Noodlification: remove duplicities

### PR DESCRIPTION
**Pull request by vhavlena**
Remove duplicities from noodlification. Now it should contain unique noodles only.

### CODE REVIEW

**Review by Adda0 — APPROVED** [review]
The changes seem sound to me.

File: src/strings/tests-nfa-noodlification.cc [hunk 1]
```diff
@@ -449,6 +449,28 @@ TEST_CASE("Mata::Nfa::SegNfa::noodlify_for_equation() both sides") {
...
+                {{x, {0, 0} }, {x, {1, 1} }},
+            } );
+        SegNfa::NoodleSubstSequence noodles = SegNfa::noodlify_for_equation(
+            std::vector<std::shared_ptr<Nfa>>{std::make_shared<Nfa>(x), std::make_shared<Nfa>(y) }, 
+            std::vector<std::shared_ptr<Nfa>>{std::make_shared<Nfa>(z), std::make_shared<Nfa>(u)});
+        CHECK(noodles.size() == 1);
+        for(size_t i = 0; i < noodles.size(); i++) {
+            for(size_t j = 0; j < noodles[i].size(); j++) {
+                CHECK(noodles[i][j].second == res[i][j].second);
+                CHECK(are_equivalent(*noodles[i][j].first.get(), res[i][j].first, nullptr));
```
Comment (jurajsic) [inline src/strings/tests-nfa-noodlification.cc:470]: You do not need to do `.get()` here, shared pointers act as pointers, so `*noodles[i][j].first` is enough.

**Review by jurajsic — APPROVED** [review]
Looks ok to me.

File: src/strings/tests-nfa-noodlification.cc (hunk 1 above)
Comment (Adda0) [reply src/strings/tests-nfa-noodlification.cc:470]: You are right. It is a syntactic sugar.

@vhavlena Do you want to address it, or will you only keep it in mind and leave this code be? If you choose the latter, I believe we can merge the PR.

Comment (vhavlena) [reply src/strings/tests-nfa-noodlification.cc:470]: Yes, comparison of shared pointers is done via comparison of .get() addresses. Let's merge it.

**Pull request merged**